### PR TITLE
feat(core): migrate command can succeed if no migrations.json using flag --if-exists

### LINKED ISSUE #14131: Allow flag in nx migrate --run-migrations to skip if migrations.json does not exist instead of failing
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

I want to use Renovate to automate the process of `nx migrate`, in the package [`@nx-squeezer/workspace`](https://github.com/nx-squeezer/squeezer/tree/main/packages/workspace/src/generators/renovate). Right now there is an integration issue in the way it should check if `migrations.json` and later run it.

The issue is well explained here https://github.com/renovatebot/renovate/issues/7767

## Motivation

Simplify the usage of `nx migrate`  in Renovate runners

## Suggested Implementation

Similar as with `npm run`, add a new flag `--if-present` that would not cause the `nx` command to fail if the `migrations.json` file does not exist

### PR DESCRIPTION
**Pull request by samuelfernandez**
## Current Behavior

Running `nx migrate --run-migrations` fails if `migrations.json` does not exist. This is not suitable for CI environment as Renovate

## Expected Behavior

With a new flag `--if-present`, the command can succeed if the file doesn't exist.

## Related Issue(s)

Fixes #14131

### CODE REVIEW

**Comment by samuelfernandez** [comment]
@AgentEnder This will need an empty commit from your side to let the CI run

File: packages/nx/src/command-line/nx-commands.ts [hunk 1]
```diff
@@ -281,7 +281,7 @@ export const commandsObject = yargs
     command: 'migrate [packageAndVersion]',
     describe: `Creates a migrations file or runs migrations from the migrations file.
   - Migrate packages and create migrations.json (e.g., nx migrate @nrwl/workspace@latest)
-  - Run migrations (e.g., nx migrate --run-migrations=migrations.json)`,
+  - Run migrations (e.g., nx migrate --run-migrations=migrations.json). Use flag --if-exists to run migrations only if the migrations file exists.`,
     builder: (yargs) =>
       linkToNxDevAndExamples(withMigrationOptions(yargs), 'migrate'),
```
Comment (AgentEnder) [inline packages/nx/src/command-line/nx-commands.ts:286]: The option needs to be added in to the `withMigrationOptions` helper, the yargs builder object is what scopes out the --help text and similar.

**Review by AgentEnder — CHANGES_REQUESTED** [review]

**Comment by AgentEnder** [comment]
There are also some docs changes, run `yarn documentation` and push them up.

**Comment by samuelfernandez** [comment]
> There are also some docs changes, run `yarn documentation` and push them up.

@AgentEnder should be fixed now! Can't do local dev because I use Windows, currently it is blocked. Did the changes manually. This would need another empty commit for the pipeline to run :shrug: Thanks for the time to review! 😄

**Comment by AgentEnder** [comment]
> Can't do local dev because I use Windows

FWIW I use windows for 95% of my work, the repo works perfectly **_under WSL_** if you are on win 10/11. Outside of WSL yeah, there are some errors. I'll push up a new commit to kick off CI and include any docs changes

**Comment by samuelfernandez** [comment]
> > Can't do local dev because I use Windows
> 
> 
> 
> FWIW I use windows for 95% of my work, the repo works perfectly **_under WSL_** if you are on win 10/11. Outside of WSL yeah, there are some errors. I'll push up a new commit to kick off CI and include any docs changes

Good to know! I tried with WSL, but hit this bug https://github.com/nrwl/nx/issues/14403 Anyway, thanks for the follow up 😀

File: packages/nx/src/command-line/migrate.spec.ts [hunk 2]
```diff
@@ -745,6 +745,18 @@ describe('Migration', () => {
       });
     });
 
+    it('should work for running migrations', () => {
+      const r = parseMigrationsOptions({
+        runMigrations: '',
+        ifExists: '',
```
Comment (samuelfernandez) [inline packages/nx/src/command-line/migrate.spec.ts:751]:
```suggestion
        ifExists: true,
```

**Comment by samuelfernandez** [comment]
@AgentEnder requested changes should be applied 😄

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**